contracts/multisig/src/key.rs

```diff
@@ -368,6 +375,26 @@ mod ecdsa_tests {
         );
     }
 
+    #[test]
+    fn validate_ecdsa_public_key_not_in_curve() {
+        // Valid compressed point format, which should produce an invalid point when decompressed
+        let invalid_compressed_point = EncodedPoint::from_bytes([
+            3, 132, 180, 161, 194, 115, 211, 43, 90, 122, 205, 26, 76, 14, 117, 209, 243, 206, 192,
+            34, 107, 93, 142, 13, 50, 95, 115, 188, 140, 82, 194, 140, 235,
+        ])
+        .unwrap();
+
+        assert!(Into::<bool>::into(
+            AffinePoint::from_encoded_point(&invalid_compressed_point).is_none()
+        ));
```

Comment:
why this is this `into` necessary?

Reply:
in this case `is_none()` doesn't return a bool, it returns `Choice`, so I had to convert

Reply:
would this work instead?
```suggestion
        assert!(AffinePoint::try_from(&invalid_compressed_point).is_err());
```